core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -50,10 +50,11 @@ import org.apache.spark.util.{Clock, SystemClock, Utils}
 private[scheduler] class BlacklistTracker (
     private val listenerBus: LiveListenerBus,
     conf: SparkConf,
+    sc: Option[SparkContext],
```

Comment:
can this take an `ExecutorAllocationClient` instead, to avoid exposing all of `SparkContext`?  also doesn't seem like it should be an option.  If the only reason is for tests, you can use a mock in those tests.